comms/core/src/peer_manager/peer_features.rs

```diff
@@ -64,6 +64,11 @@ impl PeerFeatures {
             _ => "unknown",
         }
     }
+
+    /// Returns the bits of the PeerFeatures as an i32
+    pub fn to_i32(&self) -> i32 {
+        i32::try_from(self.bits()).unwrap_or_default()
```

Comment:
Is this used? Seems kind of dangerous as it will loose data silently. Since both i32 and u32 are 32 bits it is more correct to cast `self.bits() as i32`, which could be negative but is lossless (i.e. you can convert it back using `my_i32 as u32`. This is the correct way to convert this when using diesel, or if this is not used, remove it
```suggestion
        i32::try_from(self.bits()).unwrap_or_default()
```